pennylane/transforms/hamiltonian_expand.py

```diff
@@ -77,6 +100,25 @@ def hamiltonian_expand(tape):
         raise ValueError(
             "Passed tape must end in `qml.expval(H)`, where H is of type `qml.Hamiltonian`"
         )
-
-    hamiltonian.simplify()
-    return qml.transforms.measurement_grouping(tape, hamiltonian.ops, hamiltonian.coeffs)
+    if group:
+        hamiltonian.simplify()
+        return qml.transforms.measurement_grouping(tape, hamiltonian.ops, hamiltonian.coeffs)
+
+    # create tapes that measure the Pauli-words in the Hamiltonian
+    tapes = []
+    for ob in hamiltonian.ops:
+        with tape.__class__() as new_tape:
+            for op in tape.operations:
+                qml.apply(op)
+            qml.expval(ob)
+        new_tape = new_tape.expand(stop_at=lambda obj: True)
```

Comment:
How come you have this expansion? It seems to do nothing?

Reply:
That is done in the grouping module when the tapes are split...I wasn't sure to be honest. Can remove!

Reply:
Oh, weird. Only thing I can think of is that this causes an implicit `new_tape._update()` inside the `identity`.

Reply:
Deleted it :)